Source/Core/DolphinQt/Debugger/CodeViewWidget.cpp

```diff
@@ -703,16 +741,24 @@ void CodeViewWidget::OnCopyTargetAddress()
   if (Core::GetState() != Core::State::Paused)
     return;
 
-  const std::string code_line = PowerPC::debug_interface.Disassemble(GetContextAddress());
+  const u32 addr = GetContextAddress();
+
+  const std::string code_line = [addr] {
+    Core::CPUThreadGuard guard;
+    return PowerPC::debug_interface.Disassemble(&guard, addr);
+  }();
 
   if (!IsInstructionLoadStore(code_line))
     return;
 
-  const std::optional<u32> addr =
+  const std::optional<u32> target_addr =
       PowerPC::debug_interface.GetMemoryAddressFromInstruction(code_line);
 
   if (addr)
-    QApplication::clipboard()->setText(QStringLiteral("%1").arg(*addr, 8, 16, QLatin1Char('0')));
+  {
+    QApplication::clipboard()->setText(
+        QStringLiteral("%1").arg(*target_addr, 8, 16, QLatin1Char('0')));
+  }
```

Comment:
Shouldn't this be using `if (target_addr.has_value())`? Now it's checking if `GetContextAddress() != 0`.